qiskit_experiments/framework/composite/batch_experiment.py

```diff
@@ -69,17 +86,28 @@ def circuits(self):
                 qubit_mapping = None
             else:
                 qubit_mapping = [self._qubit_map[qubit] for qubit in expr.physical_qubits]
-            for circuit in expr.circuits():
-                # Update metadata
-                circuit.metadata = {
-                    "experiment_type": self._type,
-                    "composite_metadata": [circuit.metadata],
-                    "composite_index": [index],
-                }
-                # Remap qubits if required
-                if qubit_mapping:
-                    circuit = self._remap_qubits(circuit, qubit_mapping)
-                batch_circuits.append(circuit)
+
+            if isinstance(expr, BatchExperiment):
+                circ_groups = expr._batch_circuits()
```

Comment:
good catch! 💯 probably you need to rename this method to more general name such as composite circuit and then attach this to base composite experiment as abstract in following PR after we define the behavior of parallel transpile. Otherwise you need to care this branching for all experiment types which might be added by future PRs.